### ISSUE TITLE
[a11y-defect-2]: Benefit Letters' form focus must be set properly when users edit their address

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#96332]
### Point of contact

Trevor Pierce

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

The Benefits Letters form (1 of 2) has an inline form to edit users' address. The first checkbox is not receiving keyboard focus on first render when users press the "Edit" button. This can be confusing to screen reader users because they're dropped into the middle of a form and might inadvertently create errors for themselves. Screenshot attached below.

---
![Image](https://github.com/user-attachments/assets/1e4cf05e-6009-4d55-a38a-04eb72f70de2)

### Reproduction steps

1. Log into https://staging.va.gov/records/download-va-letters/letters/confirm-address with a test user. I used 54.
2. Turn on the screen reader of your choosing
3. Click the "Edit" link to switch the text address into the form
4. Verify focus lands on one of the address fields on first run
5. Canceling and editing the form again causes focus to be set properly. That's what makes this issue so tricky.
...

### Proposed solution or next steps

Focus needs to be set on the first checkbox reliably. This may be a timing issue or in how the form is being rendered. We might be able to look at an `autofocus` attribute for the checkbox. Normally I'm **not** in favor of this, but we need a reliable, reproducible way to set focus when users enter the form.

### References, articles, or WCAG support

1. Understanding [SC 1.3.2](https://w3.org/TR/WCAG22#meaningful-sequence): Meaningful Sequence (Level A)
2. [MDN: autofocus](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus)

### Type of issue

- [ ] Axe-core
- [x] Screenreader | assistive tech & device support
- [x] Keyboard
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [x] Focus
- [ ] Headings
- [ ] Content style guide
- [ ] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by dysbo** [comment]
Waiting on AWS access - Trevor has a ticket in.

**Comment by 1Copenut** [comment]
@dysbo & @salihaghaffar I've reassigned this ticket to Forms Library Support and notified their channel on Slack.

**Comment by dysbo** [comment]
Need to escalate with Forms Library to ensure there is movement before we can move this along.

**Comment by 1Copenut** [comment]
@dysbo Filed a support request with Platform team at EOD December 3rd.

**Comment by salihaghaffar** [comment]
@dysbo will follow up.

**Comment by dysbo** [comment]
I dropped a message into [#veteran-facing-forms](https://dsva.slack.com/archives/C044AGZFG2W/p1741191251960959) in Slack, as I found this was where @1Copenut had initiated the conversation, though he didn't get a response in that channel.  I also have not received a response yet but we will give it some days and then reassess our options.

**Comment by dysbo** [comment]
This may actually be solved through the fix for https://github.com/department-of-veterans-affairs/va.gov-team/issues/116266.  Should be focusing correctly for all users once the letters and documents work is released.
